core/src/main/java/org/elasticsearch/common/Strings.java

```diff
@@ -521,16 +534,29 @@ public static String cleanPath(String path) {
         final int len = chars.length;
         int start = 0;  // starting index in chars of the current substring.
         int pos = 0;    // current index in chars.
+        int whitespaceStart = -1; // the position of the start of the current whitespace, -1 if not on whitespace
```

Comment:
Instead of this, I think you could do just like with start, and have an `end` var that marks the current end of the token, and do not increment it when you find whitespace?